catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/provider/SolrProviderTestBase.java

```diff
@@ -0,0 +1,195 @@
+
+  static final int ALL_RESULTS = -1;
+
+  static final String DEFAULT_TEST_ESCAPE = "\\";
+
+  static final String DEFAULT_TEST_SINGLE_WILDCARD = "?";
+
+  static final String DEFAULT_TEST_WILDCARD = "*";
+
+  protected static SolrCatalogProvider provider = SolrProviderTest.getProvider();
+
+  protected static SolrFilterBuilder filterBuilder = new SolrFilterBuilder();
+
+  private static final int TEST_METHOD_NAME_INDEX = 3;
+
+  private static final int ONE_SECOND = 1;
+
+  private static final int TIME_STEP_10SECONDS = 10 * ONE_SECOND;
+
+  private static final int A_LITTLE_WHILE = TIME_STEP_10SECONDS;
+
+  protected static void deleteAll() throws IngestException, UnsupportedQueryException {
+    deleteAll(TEST_METHOD_NAME_INDEX);
+  }
+
+  protected static void deleteAll(int methodNameIndex)
+      throws IngestException, UnsupportedQueryException {
+    messageBreak(Thread.currentThread().getStackTrace()[methodNameIndex].getMethodName() + "()");
+
+    QueryImpl query;
+    SourceResponse sourceResponse;
+    query = new QueryImpl(filterBuilder.attribute(Metacard.ID).is().like().text("*"));
+    query.setPageSize(ALL_RESULTS);
+    sourceResponse = provider.query(new QueryRequestImpl(query));
+
+    List<String> ids = new ArrayList<>();
+    for (Result r : sourceResponse.getResults()) {
+      ids.add(r.getMetacard().getId());
+    }
+
+    LOGGER.info("Records found for deletion: {}", ids);
+
+    provider.delete(new DeleteRequestImpl(ids.toArray(new String[ids.size()])));
+
+    LOGGER.info("Deletion complete. -----------");
+  }
+
+  QueryRequest quickQuery(Filter filter) {
+    return new QueryRequestImpl(new QueryImpl(filter));
+  }
+
+  void queryAndVerifyCount(int count, Filter filter) throws UnsupportedQueryException {
+    Query query = new QueryImpl(filter);
+    QueryRequest request = new QueryRequestImpl(query);
+    SourceResponse response = provider.query(request);
+
+    assertEquals(count, response.getResults().size());
+  }
+
+  protected DeleteResponse delete(String identifier) throws IngestException {
+    return delete(new String[] {identifier});
+  }
+
+  protected DeleteResponse delete(String[] identifier) throws IngestException {
+    return provider.delete(new DeleteRequestImpl(identifier));
+  }
+
+  protected UpdateResponse update(String id, Metacard metacard) throws IngestException {
+    String[] ids = {id};
+    return update(ids, Collections.singletonList(metacard));
+  }
+
+  protected UpdateResponse update(String[] ids, List<Metacard> list) throws IngestException {
+    return provider.update(new UpdateRequestImpl(ids, list));
+  }
+
+  protected CreateResponse create(Metacard metacard) throws IngestException {
+    return create(Collections.singletonList(metacard));
+  }
+
+  protected CreateResponse create(List<Metacard> metacards) throws IngestException {
+    return createIn(metacards, provider);
+  }
+
+  void addMetacardWithModifiedDate(DateTime now) throws IngestException {
```

Comment:
✏️ Feels weird that the argument is called `now` but could be any `DateTime` really.
❓ Also, should we add something about the location being Flagstaff in the method name, or actually take in a location?